project/activities/views.py

```diff
@@ -7,6 +7,73 @@
 
 from .forms import ActivityModelForm
 from .models import Activity
+from .serializers import ActivitySerializer
+
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+from rest_framework import status
+
+
+@api_view(['GET', 'POST'])
+def activity_list(request):
+    """
+    Circle member(s) can create a Circle Activity via a POST request
+    Circle members and coordinators can view all Circle Activities via a GET request
+    """
+    circle_id = Activity.objects.values_list("circle_id", flat=True).first()
+    circle = Circle.objects.get(id=circle_id)
+
+    if request.user in circle.companions:
+
+        if request.method == 'GET':
+            activities = Activity.objects.all()
```

Comment:
This would seem to be a heavy database query, since it could get a cursor to every activity for every circle. The activities should be scoped to the circle.